alternator/executor.cc

```diff
@@ -1358,11 +1358,27 @@ mutation put_or_delete_item::build(schema_ptr schema, api::timestamp_type ts) co
 // s_default_timeout is overwritten in alternator::controller::start_server()
 // based on the "alternator_timeout_in_ms" configuration parameter.
 db::timeout_clock::duration executor::s_default_timeout = 10s;
+db::timeout_clock::duration executor::default_getitem_timeout = 3s;
+db::timeout_clock::duration executor::default_putitem_nonlwt_timeout = 2s;
+db::timeout_clock::duration executor::default_putitem_lwt_timeout = 5s;
+db::timeout_clock::duration executor::default_query_timeout = 10s;
 void executor::set_default_timeout(db::timeout_clock::duration timeout) {
     s_default_timeout = timeout;
 }
-db::timeout_clock::time_point executor::default_timeout() {
-    return db::timeout_clock::now() + s_default_timeout;
+void executor::set_default_getitem_timeout(db::timeout_clock::duration timeout) {
+    default_getitem_timeout = timeout;
+}
+void executor::set_default_putitem_nonlwt_timeout(db::timeout_clock::duration timeout) {
+    default_putitem_nonlwt_timeout = timeout;
+}
+void executor::set_default_putitem_lwt_timeout(db::timeout_clock::duration timeout) {
+    default_putitem_lwt_timeout = timeout;
+}
+void executor::set_default_query_timeout(db::timeout_clock::duration timeout) {
+    default_query_timeout = timeout;
+}
+db::timeout_clock::time_point executor::default_timeout(db::timeout_clock::duration timeout) {
+    return db::timeout_clock::now() + timeout;
```

Comment:
This function, in this state, should no longer be called "default_timeout()" - it doesn't use any default, it just takes the parameter and adds it to now().
Maybe we don't even need such a function if it's just now() + timeout?